hpx/runtime/threads/detail/create_work.hpp

```diff
@@ -81,12 +81,16 @@ namespace hpx { namespace threads { namespace detail
         // Pass critical priority from parent to child.
         if (self)
         {
-            if (thread_priority_critical == threads::get_self_id()->get_priority())
-                data.priority = thread_priority_critical;
+            if (thread_priority_high_recursive ==
+                threads::get_self_id()->get_priority())
+            {
+                data.priority = thread_priority_high_recursive;
```

Comment:
Should this check that the promoted thread is "default" or normal. If I have a child task of a high_riority_recursive and I launch it as low priority, I think it should go low and not be promoted to high.